nvm.sh

```diff
@@ -3481,11 +3483,14 @@ nvm_auto() {
       nvm install >/dev/null
     fi
   elif [ "_$NVM_MODE" = '_use' ]; then
-   VERSION="$(nvm_resolve_local_alias default 2>/dev/null || nvm_echo)"
-    if [ -n "$VERSION" ]; then
-      nvm use --silent "$VERSION" >/dev/null
-    elif nvm_rc_version >/dev/null 2>&1; then
-      nvm use --silent >/dev/null
+    if [ "_$NVM_CURRENT" = '_none' ] || \
+       [ "_$NVM_CURRENT" = '_system' ]; then
```

Comment:
Basically this means that the current behavior of "if I source nvm.sh again, my node version gets reset to the default" would go away.

That's surely a breaking change. I can't decide if it's better behavior or not.

The rationale for force-setting the default version when `$NVM_MODE` is "use" is that we want to ensure that things work as expected after it's sourced.

What would you think about changing this so that when it's not "none" or "system", it at least still invokes `nvm use --silent "${NVM_CURRENT}" >/dev/null`? That way, we'd ensure that any `nvm use` changes between the previous nvm.sh and the new nvm.sh would be applied, but the resulting node version wouldn't have to change (which is the goal of this PR as I understand it)

Reply:
That makes sense, I will add the change.

Reply:
Added.